syntax = "proto3";

package messages.events.v1;

import "google/protobuf/any.proto";
import "google/protobuf/timestamp.proto";

message Sms {
  string id = 1;
  optional string sender = 2;
  string recipient = 3;
  string message = 4;
  SmsStatus status = 5;
  optional SmsResponse response = 6;
}

message SmsResponse {
  string id = 1;
  string account_sid = 2;
  string sid = 3;
  SmsDate sms_date = 4;
  SmsType sms_type = 5;
  int64 num_media = 6;
  int64 num_segments = 7;
  SmsPrice price = 8;
  SmsStatus status = 9;
  map<string, google.protobuf.Any> subresource_uris = 10;
  string uri = 11;
  string sms_id = 12;
  optional string messaging_service_sid = 13;
  optional string error_code = 14;
  optional string error_message = 15;
}

message SmsDate {
  google.protobuf.Timestamp date_created = 1;
  google.protobuf.Timestamp date_sent = 2;
  google.protobuf.Timestamp date_updated = 3;
}

message SmsPrice {
  optional float price = 1;
  optional string currency = 2;
}

enum SmsType {
  OUTBOUND = 0;
  INBOUND = 1;
  OUTBOUND_API = 2;
  OUTBOUND_REPLY = 3;
  UNKNOWN = 4;
}

enum SmsStatus {
  ACCEPTED = 0;
  SCHEDULED = 1;
  CANCELED = 2;
  QUEUED = 3;
  SENDING = 4;
  SENT = 5;
  FAILED = 6;
  DELIVERED = 7;
  UNDELIVERED = 8;
  RECEIVING = 9;
  READ = 10;
  PENDING = 11;
}
